service/controller/chart/v1/resource/release/update.go

```diff
@@ -2,20 +2,108 @@ package release
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/controller"
+	"k8s.io/helm/pkg/helm"
 )
 
+func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	releaseState, err := toReleaseState(updateChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if releaseState.Name != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating release %#q", releaseState.Name))
+
+		tarballURL := key.TarballURL(customObject)
+		tarballPath, err := r.helmClient.PullChartTarball(ctx, tarballURL)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		defer func() {
+			err := r.fs.Remove(tarballPath)
+			if err != nil {
+				r.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("deletion of %#q failed", tarballPath), "stack", fmt.Sprintf("%#v", err))
+			}
+		}()
+
+		err = r.helmClient.EnsureTillerInstalled(ctx)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		values, err := json.Marshal(releaseState.Values)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		// We need to pass the ValueOverrides option to make the update process
+		// use the default values and prevent errors on nested values.
+		err = r.helmClient.UpdateReleaseFromTarball(ctx, releaseState.Name, tarballPath, helm.UpdateValueOverrides(values))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updated release %#q", releaseState.Name))
+
+	}
+
+	return nil
+}
+
 func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*controller.Patch, error) {
 	create, err := r.newCreateChange(ctx, obj, currentState, desiredState)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
+	update, err := r.newUpdateChange(ctx, obj, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
 
 	patch := controller.NewPatch()
 	patch.SetCreateChange(create)
-	// patch.SetUpdateChange(update)
+	patch.SetUpdateChange(update)
 
 	return patch, nil
 }
+
+func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	currentReleaseState, err := toReleaseState(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	desiredReleaseState, err := toReleaseState(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding out if the %#q release has to be updated", desiredReleaseState.Name))
+
+	if isReleaseInTransitionState(currentReleaseState) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("the %#q release is in status %#q and cannot be updated", desiredReleaseState.Name, currentReleaseState.Status))
+		return nil, nil
```

Comment:
@tomahawk28 Good call. This is probably the most complex part of the resource but it can be simplified.

I changed the code to log and return early if the Helm release is currently transitioning.